src/net.c

```diff
@@ -174,6 +174,7 @@ struct mg_connection *mg_connect(struct mg_mgr *mgr, const char *url,
   } else {
     LIST_ADD_HEAD(struct mg_connection, &mgr->conns, c);
     c->is_udp = (strncmp(url, "udp:", 4) == 0);
+    c->fd = S2PTR(INVALID_SOCKET);
```

Comment:
I'd do

```c
  c->fd = (void *) (size_t) -1;  // Set to INVALID_SOCKET
```

Reply:
But INVALID_SOCKET is platform dependent:
MG_ARCH_FREERTOS_TCP defines it to FREERTOS_INVALID_SOCKET
MG_ARCH_WIN32 && MG_ENABLE_WINSOCK and MG_ARCH_TIRTOS seem to have a built-in value
and else it is defined to -1, so just using -1 might fail in 
https://github.com/cesanta/mongoose/blob/4271aa5b1f97f09cddf8688977ef43bd541f1887/mongoose.c#L4248-L4259

Reply:
Hardcoding to -1 seems fine to me